pkg/services/searchV2/service.go

```diff
@@ -0,0 +1,106 @@
+package searchV2
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/services/searchV2/extract"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+type StandardSearchService struct {
+	sql *sqlstore.SQLStore
+}
+
+func ProvideService(sql *sqlstore.SQLStore) SearchService {
+	return &StandardSearchService{
+		sql: sql,
+	}
+}
+
+type dashMeta struct {
+	id        int64
+	is_folder bool
+	dash      *extract.DashboardInfo
+}
+
+func (s *StandardSearchService) DoDashboardQuery(ctx context.Context, query DashboardQuery) *backend.DataResponse {
+	rsp := &backend.DataResponse{}
+
+	dash, err := loadDashboards(ctx, s.sql)
+	if err != nil {
+		rsp.Error = err
+		return rsp
+	}
+
+	rsp.Frames = append(rsp.Frames, metaToFrame(dash))
+
+	return rsp
+}
+
+func loadDashboards(ctx context.Context, sql *sqlstore.SQLStore) ([]dashMeta, error) {
+	meta := make([]dashMeta, 0, 200)
+
+	// key will allow name or uid
+	lookup := func(key string) *extract.DatasourceInfo {
+		return nil // TODO!
+	}
+
+	err := sql.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		// alertRules := make([]*ngmodels.AlertRule, 0)
+		// q := "SELECT * FROM alert_rule WHERE org_id = ?"
+		// params := []interface{}{query.OrgID}
+
+		res, err := sess.Query("SELECT id,is_folder,data FROM dashboard") // TODO -- limit by orgID!!!
+		if err != nil {
+			return err
+		}
+
+		for _, row := range res {
+			// id := row["id"]
+			// is_folder := row["is_folder"]
```

Comment:
how do we parse the "is_folder" and "id" -- they are `[]byte`... but need int/bool :thinking: